### PR TITLE
feat: add new deployment events and update console

### LINKED ISSUE #300: Add dedicated scaling events to the timeline

### PR DESCRIPTION
**Pull request by wesbillman**
Fixes #300 
Screenshot 2023-09-21 at 8 40 27 AM
Screenshot 2023-09-21 at 8 41 53 AM
Screenshot 2023-09-21 at 8 41 58 AM

### CODE REVIEW

**Comment by wesbillman** [comment]
@alecthomas you'll probably want to skim through this one when you have a minute. Lemme know if you want anything changed or updated here.

**Comment by dhanji** [comment]
gorgeous!

File: backend/controller/console.go [hunk 1]
```diff
@@ -186,87 +186,6 @@ func (c *ConsoleService) StreamEvents(ctx context.Context, req *connect.Request[
...
-		RequestName:    requestName,
-		TimeStamp:      timestamppb.New(event.Time),
-		LogLevel:       event.Level,
-		Attributes:     event.Attributes,
-		Message:        event.Message,
-		Error:          event.Error.Ptr(),
-	}
-}
-
-func deploymentEventToDeployment(event *dal.DeploymentEvent) *pbconsole.Deployment {
```
Comment (alecthomas) [inline backend/controller/console.go:232]: No more helpers?

File: backend/controller/internal/dal/dal.go [hunk 2]
```diff
@@ -621,10 +612,33 @@ func (p *postgresClaim) Runner() Runner { return p.runner }
 
 // SetDeploymentReplicas activates the given deployment.
 func (d *DAL) SetDeploymentReplicas(ctx context.Context, key model.DeploymentName, minReplicas int) error {
-	err := d.db.SetDeploymentDesiredReplicas(ctx, key, int32(minReplicas))
+	// Start the transaction
+	tx, err := d.db.Begin(ctx)
```
Comment (alecthomas) [inline backend/controller/internal/dal/dal.go:616]: Noice.

File: backend/controller/internal/dal/dal.go [hunk 3]
```diff
@@ -621,10 +612,33 @@ func (p *postgresClaim) Runner() Runner { return p.runner }
 
 // SetDeploymentReplicas activates the given deployment.
 func (d *DAL) SetDeploymentReplicas(ctx context.Context, key model.DeploymentName, minReplicas int) error {
```
Comment (alecthomas) [inline backend/controller/internal/dal/dal.go:614]: I think to make `tx.CommitOrRollback()` work correctly, you'll want to change the return signature to `(err error)`.

File: cmd/ftl/cmd_update.go [hunk 4]
```diff
@@ -13,7 +13,7 @@ import (
 
 type updateCmd struct {
 	Replicas   int32                `short:"n" help:"Number of replicas to deploy." default:"1"`
-	Deployment model.DeploymentName `arg:"" help:"Deployment to kill."`
+	Deployment model.DeploymentName `arg:"" help:"Deployment to update."`
```
Comment (alecthomas) [inline cmd/ftl/cmd_update.go:16]: Whoops!

File: console/client/src/features/timeline/TimelineIcon.tsx [hunk 5]
```diff
@@ -18,8 +18,10 @@ export const TimelineIcon = ({ event }: Props) => {
           <PhoneArrowDownLeftIcon className={`${style}`} />
         )
       }
-      case 'deployment':
-        return <RocketLaunchIcon className={`${style}`} />
+      case 'deploymentCreated':
```
Comment (alecthomas) [inline console/client/src/features/timeline/TimelineIcon.tsx:21]: Can we use the proto enum types here?

**Pull request merged**